src/defaults.js

```diff
@@ -7,12 +7,36 @@ const { defaultOptions } = require("./utils");
  */
 const TUS_CHUNK_SIZE = (1 << 22) * 10;
 
+/**
+ * Indicates what the default chunk size multiplier is.
+ */
+const DEFAULT_TUS_CHUNK_SIZE_MULTIPLIER = 1;
```

Comment:
default in skynet-js is different, any reason to set it to one here?

https://github.com/SkynetLabs/skynet-js/blob/master/src/upload.ts#L26

```suggestion
const DEFAULT_TUS_CHUNK_SIZE_MULTIPLIER = 3;
```

when porting the code I would assume we're going to only adjust what's not platform compatible but leave everything else to simplify maintenance

Reply:
Hello @kwypchlo,

Now for the current problem when I set the constant like this:
`DEFAULT_TUS_CHUNK_SIZE_MULTIPLIER = 3;`
then "parallelUploads" is only executed if the file size is **over "125829120", all file sizes smaller than "125829120"** lead to an error.

Example with approx. **52mb file size** with error:
```
Failed because: Error: Expected parameter 'totalSize' to be greater than the size of a chunk ('125829120'), was type 'number', value '52413226'
Error: Expected parameter 'totalSize' to be greater than the size of a chunk ('125829120'), was type 'number', value '52413226'
    at validationError (D:\z3\2\nodejs-skynet\src\utils_validation.js:23:10)
    at throwValidationError (D:\z3\2\nodejs-skynet\src\utils_validation.js:36:9)
    at splitSizeIntoChunkAlignedParts (D:\z3\2\nodejs-skynet\src\utils_testing.js:58:5)
    at splitSizeIntoParts (D:\z3\2\nodejs-skynet\src\upload.js:119:14)
    at Upload._startParallelUpload (D:\z3\2\nodejs-skynet\node_modules\@skynetlabs\tus-js-client\lib.es5\upload.js:390:19)
    at D:\z3\2\nodejs-skynet\node_modules\@skynetlabs\tus-js-client\lib.es5\upload.js:359:18
```
This error comes because the example 52MB file is smaller than "125829120".

Because the constant **"TUS_CHUNK_SIZE"** is fixed at about "41943040", and file upload is **"41943040 * 3 = 125829120"**.

So we should set it like this:
`DEFAULT_TUS_CHUNK_SIZE_MULTIPLIER = 1;`
so that files from a size of "41943040" with "parallelUploads" also work, and thus the example with 52mb file also works great, and also without errors.

In my opinion it should work normally, why the constants in "SkynetLabs/skynet-js" are set so high as default, I don't understand.

Reply:
I think it fails because you missed to change this line with the condition

skynet-js https://github.com/SkynetLabs/skynet-js/blob/master/src/upload.ts#L130
nodejs-skynet https://github.com/SkynetLabs/nodejs-skynet/blob/master/src/upload.js#L26

let's amend this line, restore skynet-js defaults and see if that helps

Reply:
Hello @kwypchlo,

Thanks for the change
I must have overlooked it, I changed it right away.

The problem persists that when I change the constants to:
```
const DEFAULT_TUS_CHUNK_SIZE_MULTIPLIER = 3 ;
const DEFAULT_TUS_PARALLEL_UPLOADS = 2 ;
```

1. Problem is that if `DEFAULT_TUS_PARALLEL_UPLOADS = 2` is set, "parallelUploads" is always active for every upload from approx. `42mb * 3 = 125829120`.

With all upload files between approx. 42mb - 125mb this error message appears:
```
Error: Expected parameter 'totalSize' to be greater than the size of a chunk ('125829120'), was type 'number', value '52413226'
    at validationError (D:\z6\1\nodejs-skynet\src\utils_validation.js:23:10)
    at throwValidationError (D:\z6\1\nodejs-skynet\src\utils_validation.js:36:9)
    at splitSizeIntoChunkAlignedParts (D:\z6\1\nodejs-skynet\src\utils_testing.js:62:5)
    at splitSizeIntoParts (D:\z6\1\nodejs-skynet\src\upload.js:119:14)
    at Upload._startParallelUpload (D:\z6\1\nodejs-skynet\node_modules\@skynetlabs\tus-js-client\lib.es5\upload.js:390:19)
    at D:\z6\1\nodejs-skynet\node_modules\@skynetlabs\tus-js-client\lib.es5\upload.js:359:18
```

2. Problem is the error message above that is thrown by
`const DEFAULT_TUS_CHUNK_SIZE_MULTIPLIER = 3 ;`

If `DEFAULT_TUS_CHUNK_SIZE_MULTIPLIER = 1` , then "chunkSize" would only be about 42mb and not about 125mb,
**which then also brings no error messages**, even if as a default
const DEFAULT_TUS_PARALLEL_UPLOADS = 2 ;
is active.



@kwypchlo, I can't get the issue resolved, and need urgent help with the issue.
Maybe @m-cat can help here.

I myself have no problem with setting the constants to the same values ​​of "SkynetLabs/skynet-js", because I know, so that no error messages between approx if I don't want "parallelUploads" as the standard, I simply set DEFAULT_TUS_PARALLEL_UPLOADS to 1, then it works great for me and I think how it should be.

Reply:
Hello @kwypchlo,

**Here are my test instructions, how I run my test locally.**

you can test it yourself by git cloning my repo https://github.com/parajbs/nodejs-skynet and upload a file with a size of about 50mb with the command
`node scripts/upload.js path-to-50mb-file-to-upload`
The upload should work normally without "parallelUploads" and without an error message.

The second test then the same approx. 50mb file with "parallelUploads" active, with this command:
`node scripts/upload_progress.js path-to-50mb-file-to-upload`
The upload of the approx. 50mb file should now work quickly and without an error message, because **"numParallelUploads: 2"** but **"DEFAULT_TUS_CHUNK_SIZE_MULTIPLIER = 1"** is set.
**Should have worked too.**

Now for her changes:
So that you can now do the tests with your changes, you simply have to change the constants to the values ​​in "src/defaults.js":
```
const DEFAULT_TUS_CHUNK_SIZE_MULTIPLIER = 3 ;
const DEFAULT_TUS_PARALLEL_UPLOADS = 2 ;
```

Now you should both run the tests above with the same ~50mb file and you will get an error message.
If the test file is **larger than 126mb**, it works again.

Test3 command:
`node scripts/upload.js path-to-50mb-file-to-upload`

Test4 command:
`node scripts/upload_progress.js path-to-50mb-file-to-upload`

@kwypchlo, you should urgently test it locally with my repo (current code from here), after that I'm happy to change it if you still want it.